### PR TITLE
Adding rspec

### LINKED ISSUE #13: Quick question on testing
Hey @Florent2, thanks for the great project ! :+1:  . I would also like to contribute as and when time permits. 

I would like to start with adding some more tests around the code. Before starting I have a quick question. I can see the project is using the standard rails test framework, so have you decided to stick with it or is there flexibility to move to `rspec` ?

I'm more comfortable with `rspec` and if you are ok, I would rewrite the current tests and add some more with `rspec` and `factory girl`

let me know your thoughts

cheers
Sam

### PR DESCRIPTION
**Pull request by sameera207**
fixes #13  

1 - rspec added 
    
     basic `rspec` coverage added with model, request , lib and routing specs 

2 - issue fixed in `app/views/application_mailer/new_gem_admin_notification.html.erb` ti support new columns
    
      Mailer template was using old columns.

3 - `ILIKE` changed to `lower`, because its supported by both Sqlite and Postgres
   
      having `ILIKE` is not working for tests as sqlite doesnt support the syntax. now both the user inout and column value will change to lowercase to match.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -28,4 +28,10 @@ group :development, :test do
   gem 'byebug'
   gem 'web-console', '~> 2.0'
   gem 'spring'
+  gem 'sqlite3'
+
+  gem 'rspec-rails', '~> 3.0.0'
+  gem 'factory_girl_rails'
```
Comment (Florent2) [inline Gemfile]: The factory_girl_rails, database_cleaner and fake gems should be moved to the `group :test` block.

Please remove the blind and mocha gems from this block as they are no longer used.

File: app/models/rubygem.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@ def self.by_status(status, rails_version)
   end
 
   def self.search query
-    where("name ILIKE ?", "%#{query}%")
+    where("lower(name) LIKE ?", "%#{query.try(:downcase)}%")
```
Comment (Florent2) [inline app/models/rubygem.rb]: :+1:

**Comment by Florent2** [comment]
@sameera207 awesome, thanks! I've left a comment in the Gemfile.

Would be great if you can address it, and the rebase your work against the master branch to resolve the conflicts (I've just made a few changes before seeing your pull request) and squash your commits into a single one.

See http://makandracards.com/makandra/36003-recommended-git-workflow-for-feature-branches if you never did that. If you can't do it no problem, let me know and I would take care of that.

Thanks again!

**Comment by sameera207** [comment]
@Florent2 , no worries, I'll address the comments and I'll have a look at the link and send u a new PR. :)

**Comment by sameera207** [comment]
Hey @Florent2 , I addressed your comments, but still seems like there are some merge conflicts. I might be doing something wrong. could u pls have a look :)

**Comment by Florent2** [comment]
@sameera207 I've merged your branch into master and fixed the conflict. Thanks for your contribution!

**Pull request merged**